test/e2e/ocp_cluster.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"golang.org/x/exp/slices" // TODO: Use "slices" instead once go version is updated to 1.21
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Cluster Operators", Label(smoke), func() {
+	It("should be all available", func(ctx context.Context) {
+		By("checking that all cluster operators are available")
+		cos, err := clients.ConfigClient.ConfigV1().ClusterOperators().List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		for _, co := range cos.Items {
+			available := false
+			for _, condition := range co.Status.Conditions {
+				if condition.Type == configv1.OperatorAvailable {
+					Expect(condition.Status).To(Equal(configv1.ConditionTrue), "operator %s is not available", co.Name)
+					available = true
+					break
+				}
+			}
+			Expect(available).To(BeTrue(), "operator %s is not available", co.Name)
```

Comment:
Isn't this duplicating the `Expect` check that's being done on line 29? It seems like we could remove the `available` variable and all references to it.

Reply:
Oh, I see what you mean. I don't think that's thinking too much.

With that said, I found it a bit confusing that there are two `Expect`s, since we're logically checking one thing: whether or not the operator is available.

What do you think of replacing lines 29-30 with `available = (condition.Status == configv1.ConditionTrue)`? This is totally a matter of opinion, so the code is also fine as-is if you think it's more readable that way.